vfsutil/zip.go

```diff
@@ -22,24 +23,24 @@ import (
 
 // NewZipVFS downloads a zip archive from a URL (or fetches from the local cache
 // on disk) and returns a new VFS backed by that zip archive.
-func NewZipVFS(ctx context.Context, url string, onFetchStart, onFetchFailed func(), evictOnClose bool) (*ArchiveFS, error) {
-	request, err := http.NewRequest("HEAD", url, nil)
+func NewZipVFS(ctx context.Context, urlString string, onFetchStart, onFetchFailed func(), evictOnClose bool) (*ArchiveFS, error) {
```

Comment:
It's kind of weird to use `urlString` and `urlStruct` naming. I would leave `url` as-is and rename `urlStruct` to just `u`

Reply:
What do you suggest for naming now that the the `url` package name conflicts with the local variable `url`?

Reply:
Err, hang on, I need to double check the naming here.

Reply:
it won't conflict if you do my `withoutAuth` refactor suggested below

Reply:
Like this?

```go
func withoutAuth(urlString string) (string, error) {
	u, err := url.Parse(urlString)
...
```